src/unix/solarish/mod.rs

```diff
@@ -2856,15 +2874,19 @@ extern "C" {
     pub fn nl_langinfo(item: ::nl_item) -> *mut ::c_char;
 
     #[cfg_attr(target_os = "illumos", link_name = "__xnet_bind")]
+    #[cfg_attr(target_os = "solaris", link_name = "__xnet_bind")]
     pub fn bind(socket: ::c_int, address: *const ::sockaddr, address_len: ::socklen_t) -> ::c_int;
 
     pub fn writev(fd: ::c_int, iov: *const ::iovec, iovcnt: ::c_int) -> ::ssize_t;
     pub fn readv(fd: ::c_int, iov: *const ::iovec, iovcnt: ::c_int) -> ::ssize_t;
 
     #[cfg_attr(target_os = "illumos", link_name = "__xnet_sendmsg")]
+    #[cfg_attr(target_os = "solaris", link_name = "__xnet_sendmsg")]
     pub fn sendmsg(fd: ::c_int, msg: *const ::msghdr, flags: ::c_int) -> ::ssize_t;
     #[cfg_attr(target_os = "illumos", link_name = "__xnet_recvmsg")]
+    #[cfg_attr(target_os = "solaris", link_name = "__xnet_recvmsg")]
```

Comment:
For cases where the `link_name` now applies to both illumos and Solaris, there should be no need for the `cfg_attr` test while in `unix/solarish/mod.rs`